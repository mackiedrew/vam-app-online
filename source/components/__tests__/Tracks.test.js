import Subject from "../Tracks";
jest.mock("../../containers/AudioManager", () => "AudioManager");
jest.mock("../../containers/Track", () => "Track");

const mockPropsNoTracks = {
  view: { start: 0, end: 10 },
  trackList: {}
};

const mockPropsWithTracks = {
  view: { start: 0, end: 10 },
  trackList: { track1: {}, track2: {} }
};

describe("<Tracks />", () => {
  describe("renders without crashing", () => {
    it("without tracks", () => {
      shallow(<Subject {...mockPropsNoTracks} />);
    });
    it("with tracks", () => {
      shallow(<Subject {...mockPropsWithTracks} />);
    });
  });

  describe("renders correctly", () => {
    it("without tracks", () => {
      const tree = renderer.create(<Subject {...mockPropsNoTracks} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("with tracks", () => {
      const tree = renderer
        .create(<Subject {...mockPropsWithTracks} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
