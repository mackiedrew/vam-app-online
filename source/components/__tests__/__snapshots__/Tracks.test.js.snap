// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tracks /> renders correctly with tracks 1`] = `
<section
  className="tracks"
>
  <div
    className="time-bar"
  >
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
  </div>
  <AudioManager />
  <Track
    id="track1"
  />
  <Track
    id="track2"
  />
</section>
`;

exports[`<Tracks /> renders correctly without tracks 1`] = `
<section
  className="tracks"
>
  <div
    className="time-bar"
  >
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
    <div
      className="slice"
    >
      00s:000
    </div>
  </div>
  <AudioManager />
  <div
    className="no-tracks"
  >
    no tracks
  </div>
</section>
`;
